x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichPlugin.java

```diff
@@ -120,13 +121,17 @@ public EnrichPlugin(final Settings settings) {
 
     @Override
     public List<NamedWriteableRegistry.Entry> getNamedWriteables() {
-        return Collections.singletonList(new NamedWriteableRegistry.Entry(MetaData.Custom.class, EnrichMetadata.TYPE,
-            EnrichMetadata::new));
+        return List.of(
+            new NamedWriteableRegistry.Entry(MetaData.Custom.class, EnrichMetadata.TYPE, EnrichMetadata::new),
+            new NamedWriteableRegistry.Entry(NamedDiff.class, EnrichMetadata.TYPE,
```

Comment:
without this line, a cluster state with an enrich policy cannot be published to other nodes.